### PR TITLE
SOLR-16353: SplitShardCmd.checkDiskSpace can fail for transient core

### PR DESCRIPTION
**Pull request by heythm**
https://issues.apache.org/jira/browse/SOLR-16353

TODOs:
- [x] set default value for `solr.shardSplit.checkDiskSpace`

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cloud/api/collections/SplitShardCmd.java [hunk 1]
```diff
@@ -811,6 +811,10 @@ public static void checkDiskSpace(
       SolrIndexSplitter.SplitMethod method,
       SolrCloudManager cloudManager)
       throws SolrException {
+    // check that the system property is enabled. It should not be disabled by default.
+    if (!System.getProperty("solr.shardSplit.checkDiskSpace", "true").equals("true")) {
```
Comment (heythm) [inline solr/core/src/java/org/apache/solr/cloud/api/collections/SplitShardCmd.java:815]: looking for a more elegant way to set default value and check

File: solr/core/src/java/org/apache/solr/cloud/api/collections/SplitShardCmd.java [hunk 2]
```diff
@@ -811,6 +811,10 @@ public static void checkDiskSpace(
       SolrIndexSplitter.SplitMethod method,
       SolrCloudManager cloudManager)
       throws SolrException {
+    // check that the system property is enabled. It should not be disabled by default.
+    if (System.getProperty("solr.shardSplit.checkDiskSpace").equals("true")) {
```
[the remark this answers is not in the document]
Comment (heythm) [reply solr/core/src/java/org/apache/solr/cloud/api/collections/SplitShardCmd.java]: @sonatype-lift ignore

**Review by dsmiley — APPROVED** [review]
LGTM.  I don't think a test is necessary; it's simple enough.

I do wonder if the setting should be a parameter on the request?  But I'm doubtful a client will toggle it depending on the collection or whatever.  WDYT?

Is there a reference to this disk check in the reference guide?  If so, definitely add a note on how to disable it.  If not... consider adding it.

And add a CHANGES.txt entry under the "Improvement" category for 9.1

**Comment by dsmiley** [comment]
Since it's a known issue that this check doesn't make sense for HDFS, we could document that in the ref guide for an HDFS page.

**Comment by heythm** [comment]
@dsmiley - About toggling this parameter through client requests: It depends on the use cases it will help. Would it be just to reduce the noise of the warnings or for a real utility? If we do not see other use cases to solve, it could be implemented when they are identified so as not to add a non-useful parameter.

File: solr/solr-ref-guide/modules/deployment-guide/pages/solr-on-hdfs.adoc [hunk 3]
```diff
@@ -246,6 +246,17 @@ s|Required |Default: none
 +
 The Kerberos principal that Solr should use to authenticate to secure Hadoop; the format of a typical Kerberos V5 principal is: `primary/instance@realm`.
 
+== Shard splitting settings
```
Comment (dsmiley) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solr-on-hdfs.adoc:249]: The *only* point of putting this documentation here, on the Hdfs page, would be **if** we would recommend the user to do something about this setting if they are using Hdfs.  But you flipped this setting off for HdfsDirectory and so I think it's now misplaced.  Instead, I'd specify this over in a shard split page.

And when yhou do, ensure it's clear that this is a *system property*; not a parameter or something.

File: solr/CHANGES.txt [hunk 4]
```diff
@@ -66,6 +66,8 @@ Improvements
 
 * SOLR-16323: The Docker image now uses the Solr User ID instead of the User Name, helps with non-root checks (Houston Putman)
 
+* SOLR-16353: Make SplitShardCmd#checkDiskSpace enableable through a system property. (Haythem Khiri)
```
Comment (dsmiley) [inline solr/CHANGES.txt:69]: I think *disableable*

File: solr/solr-ref-guide/modules/deployment-guide/pages/solrcloud-shards-indexing.adoc [hunk 5]
```diff
@@ -161,6 +161,8 @@ You can delete the old shard at a later time when you're ready.
 
 More details on how to use shard splitting is in the section on the Collection API's xref:shard-management.adoc#splitshard[SPLITSHARD command].
 
+When splitting a shard, a free disk space check is performed on the local file system of the leader shard. This can be disabled through the `solr.shardSplit.checkDiskSpace.enabled` system property (i.e. `-Dsolr.shardSplit.checkDiskSpace.enabled=false`). This property is disabled by default for xref:solr-on-hdfs.adoc[HDFS].
```
Comment (dsmiley) [inline solr/solr-ref-guide/modules/deployment-guide/pages/solrcloud-shards-indexing.adoc:164]: I don't think this location is a good spot.  It's an overview of a variety of things; not details of how to do something.  Just above there is a link to `shard-management.adoc` which is an appropriate place for this detail.

**Pull request merged**